### ISSUE TITLE
input mask on integer field to disallow ".

### ISSUE THREAD

**Issue by MartijnR** [kobotoolbox/enketo-express#684]

**Comment by MartijnR** [comment]
Ideally scenario:

Not be able to type . or , at all. Or a "-34-34".

**Comment by MartijnR** [comment]
https://codepen.io/MartijnR/pen/dzggGE

**Comment by MartijnR** [comment]
test values:

* -34-34
* 1.1
* 0
* 1.000000
* 1.0000001
* -1.000000

**Comment by MartijnR** [comment]
Implementation-wise this doesn't seem feasible with number inputs, because there is no way to retrieve the currently shown input value if it's not a valid number. Any attempt to circumvent this runs into an issue when a) entering negative numbers (the first character "-" by itself is an invalid number) and b) cutting-and-pasting a value.

I have come to the conclusion that without changing the type to "text" this issue is not feasible. Changing the type to text is not wise, as one looses built-in number-type functionality (spinner, numeric keyboard, keyboard number manipulation). 

Therefore, I think we should implement this pragmatically by https://github.com/kobotoolbox/enketo-express/issues/678.

**Comment by MartijnR** [comment]
we could reject any decimal (typed or pasted) immediately as if they were invalid numbers

**Comment by MartijnR** [comment]
Look into why Enketo clears 1+1 automatically from number input field

**Comment by pbowen-oc** [comment]
@MartijnR - I confirmed that the approach we discussed works for us. As a low priority item, we should probably handle decimal fields the same basic way (I have been able to type 'e' into these fields).

At some point in the future, we will want to try to handle invalid formats (eg, misplaced minus signs, multiple decimal marks, or anything invalid that can be pasted) in a way that shows the user an error without immediately clearing the value.
